server/task-distributed/src/main/java/org/apache/james/task/eventsourcing/distributed/RabbitMQTerminationSubscriber.java

```diff
@@ -0,0 +1,140 @@
+import reactor.rabbitmq.Sender;
+import reactor.rabbitmq.SenderOptions;
+
+public class RabbitMQTerminationSubscriber implements TerminationSubscriber, Startable, Closeable {
+    private static final Logger LOGGER = LoggerFactory.getLogger(RabbitMQTerminationSubscriber.class);
+    private static final Integer MAX_CHANNELS_NUMBER = 1;
+    private static final String EXCHANGE_NAME = "terminationSubscriberExchange";
+    private static final String QUEUE_NAME_PREFIX = "terminationSubscriber";
+    private static final String ROUTING_KEY = "terminationSubscriberRoutingKey";
+
+    private final JsonEventSerializer serializer;
+    private final Mono<Connection> connectionMono;
+    private final ReactorRabbitMQChannelPool channelPool;
+    private final String queueName;
+    private WorkQueueProcessor<OutboundMessage> sendQueue;
+    private Flux<Delivery> listener;
+    private Disposable sendQueueHandle;
+
+    @Inject
+    public RabbitMQTerminationSubscriber(SimpleConnectionPool simpleConnectionPool, JsonEventSerializer serializer) {
+        this.serializer = serializer;
+        this.connectionMono = simpleConnectionPool.getResilientConnection();
+        this.channelPool = new ReactorRabbitMQChannelPool(connectionMono, MAX_CHANNELS_NUMBER);
+        this.queueName = QUEUE_NAME_PREFIX + UUID.randomUUID().toString();
+    }
+
+    public void start() {
+        Sender sender = RabbitFlux.createSender(new SenderOptions()
+            .connectionMono(connectionMono)
+            .channelPool(channelPool)
+            .resourceManagementChannelMono(
+                connectionMono.map(Throwing.function(Connection::createChannel)).cache()));
+
+        Receiver receiver = RabbitFlux.createReceiver(new ReceiverOptions().connectionMono(connectionMono));
+        listener = receiver.consumeAutoAck(queueName);
+
+        sender.declareExchange(ExchangeSpecification.exchange(EXCHANGE_NAME)).block();
+        sender.declare(QueueSpecification.queue(queueName).durable(false)).block();
+        sender.bind(BindingSpecification.binding(EXCHANGE_NAME, ROUTING_KEY, queueName)).block();
+        sendQueue = WorkQueueProcessor.create();
+        sendQueueHandle = sender
+            .send(sendQueue)
+            .subscribeOn(Schedulers.elastic())
+            .subscribe();
+    }
+
+    @Override
+    public void addEvent(Event event) {
+        try {
+            byte[] payload = serializer.serialize(event).getBytes(StandardCharsets.UTF_8);
+            AMQP.BasicProperties basicProperties = new AMQP.BasicProperties.Builder().build();
+            OutboundMessage message = new OutboundMessage(EXCHANGE_NAME, ROUTING_KEY, basicProperties, payload);
+            sendQueue.onNext(message);
+        } catch (JsonProcessingException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public Stream<Event> listenEvents() {
+        return listener
+            .concatMap(this::toEvent)
+            .share()
```

Comment:
Can you explain me what is this?

Why don't we use it in other places?

Reply:
basically, all elements of the `Flux` will be duplicated for each `Subscriber`, that avoid loosing messages.

I do not know.

Reply:
Shouldn't we do that in the (mailbox) EventBus too ?

Reply:
I do not know, if we have more than a `Subscriber` for a `Flux`, we should.